[3.7.21][MATLAB] Issue: Error using load library

I am trying to control 8 Dynamixel motors via MATLAB on macOS Mojave.
I tried running the sample code read_write.m [DynamixelSDK folder]/matlab/protocol2.0,
but it gave the following error message

> Error using loadlibrary
> There was an error loading the library "libdxl_mac_c"
> dlopen(libdxl_mac_c.dylib, 6): image not found

> Error in read_write (line 48)
>    [notfound, warnings] = loadlibrary(lib_name, 'dynamixel_sdk.h', 'addheader',
>    'port_handler.h', 'addheader', 'packet_handler.h');

> Caused by:
>    Error using loaddefinedlibrary
>    dlopen(libdxl_mac_c.dylib, 6): image not found

I have also tried making the Makefile in [DynamixelSDK folder]/c/build/Mac/ to build the dynamic library, and the following error message from terminal appeared:

> make
> mkdir -p ./.objects/
> gcc -O2 -O3 -D_GNU_SOURCE -Wall -c -I../../include/dynamixel_sdk -fPIC -g -c ../../src/dynamixel_sdk/group_bulk_read.c -o .objects/group_bulk_read.o
> gcc -O2 -O3 -D_GNU_SOURCE -Wall -c -I../../include/dynamixel_sdk -fPIC -g -c ../../src/dynamixel_sdk/group_bulk_write.c -o .objects/group_bulk_write.o
> gcc -O2 -O3 -D_GNU_SOURCE -Wall -c -I../../include/dynamixel_sdk -fPIC -g -c ../../src/dynamixel_sdk/group_sync_read.c -o .objects/group_sync_read.o
> gcc -O2 -O3 -D_GNU_SOURCE -Wall -c -I../../include/dynamixel_sdk -fPIC -g -c ../../src/dynamixel_sdk/group_sync_write.c -o .objects/group_sync_write.o
> gcc -O2 -O3 -D_GNU_SOURCE -Wall -c -I../../include/dynamixel_sdk -fPIC -g -c ../../src/dynamixel_sdk/packet_handler.c -o .objects/packet_handler.o
> gcc -O2 -O3 -D_GNU_SOURCE -Wall -c -I../../include/dynamixel_sdk -fPIC -g -c ../../src/dynamixel_sdk/port_handler.c -o .objects/port_handler.o
> gcc -O2 -O3 -D_GNU_SOURCE -Wall -c -I../../include/dynamixel_sdk -fPIC -g -c ../../src/dynamixel_sdk/protocol1_packet_handler.c -o .objects/protocol1_packet_handler.o
> gcc -O2 -O3 -D_GNU_SOURCE -Wall -c -I../../include/dynamixel_sdk -fPIC -g -c ../../src/dynamixel_sdk/protocol2_packet_handler.c -o .objects/protocol2_packet_handler.o
> In file included from ../../src/dynamixel_sdk/protocol2_packet_handler.c:24:
> In file included from ../../include/dynamixel_sdk/protocol2_packet_handler.h:23:
> ../../include/dynamixel_sdk/packet_handler.h:104:25: error: conflicting types
>       for 'reboot'
> WINDECLSPEC void        reboot              (int port_num, int protocol_...
>                         ^
> /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/unistd.h:702:6: note: 
>       previous declaration is here
> int      reboot(int);
>          ^
> 1 error generated.
> make: *** [.objects/protocol2_packet_handler.o] Error 1

How should libdxl_mac_c.dylib be properly created?

--- a/c/include/dynamixel_sdk/packet_handler.h
+++ b/c/include/dynamixel_sdk/packet_handler.h
@@ -25,6 +25,11 @@
 #include "robotis_def.h"
 #include "port_handler.h"
 
+#if defined(__APPLE__)
+#undef reboot
+#define reboot rebootDXL
+#endif
+
 #define BROADCAST_ID        0xFE    // 254
 #define MAX_ID              0xFC    // 252
 
@@ -101,7 +106,11 @@ WINDECLSPEC uint16_t    pingGetModelNum     (int port_num, int protocol_version,
 WINDECLSPEC void        broadcastPing       (int port_num, int protocol_version);
 WINDECLSPEC uint8_t     getBroadcastPingResult  (int port_num, int protocol_version, int id);
 
+#if defined(__APPLE__)
+WINDECLSPEC void        rebootDXL           (int port_num, int protocol_version, uint8_t id);
+#else
 WINDECLSPEC void        reboot              (int port_num, int protocol_version, uint8_t id);
+#endif
 
 WINDECLSPEC void        clearMultiTurn      (int port_num, int protocol_version, uint8_t id);
 

--- a/c/src/dynamixel_sdk/packet_handler.c
+++ b/c/src/dynamixel_sdk/packet_handler.c
@@ -211,7 +211,11 @@ uint8_t getBroadcastPingResult(int port_num, int protocol_version, int id)
   }
 }
 
+#if defined(__APPLE__)
+void rebootDXL(int port_num, int protocol_version, uint8_t id)
+#else
 void reboot(int port_num, int protocol_version, uint8_t id)
+#endif
 {
   if (protocol_version == 1)
   {

